main.cc

```diff
@@ -1,11 +1,42 @@
-#include "marc_pop_unique_ptr/custom_unique_ptr.h"
-#include "devices/Device.h"
-#include "user/User.h"
-
-int main() {
-    smart_home::User::getInstance().controlDevicesDemo();
-    smart_home::User::getUserHouse().writeDataToFile();
-    smart_home::User::readSensorDataFromRoomDemo();
-    smart_home::User::getInstance().getDevicesStatus();
-    return 0;
-}
+#include "user/User.h"
+#include <thread>
+#include <chrono>
+#include <mutex>
+
+std::mutex readyMutex;
+bool ready = false;
+
+void statusThread() {
+    {
+        std::lock_guard<std::mutex> lock(readyMutex);
+        while (!ready) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));  
+        }
+    }
+
+    while (true) {
+        smart_home::User::getInstance().demoPrintSensorDataFromLivingRoom();
+        smart_home::User::getInstance().getDevicesStatus();
+        std::this_thread::sleep_for(std::chrono::seconds(5));
+    }
+}
+
+int main() {
+    smart_home::User::getInstance().controlDevicesDemo();
+    smart_home::User::getUserHouse().writeDataToFile();
+
+    {
+        std::lock_guard<std::mutex> lock(readyMutex);
+        ready = true;
+    }
+
+    std::thread printingThread(statusThread);

```

Comment:
you display on 2 different threads but the action of the main thread is finished by the time you create the display thread.
ControlDevicesDemo should be called during the printing thread run.